plugins/biometric/guest-js/index.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+import { invoke } from "@tauri-apps/api/primitives";
+
+export enum BiometryType {
+  None = 0,
+  // Apple TouchID or Android fingerprint
+  TouchID = 1,
+  // Apple FaceID or Android face authentication
+  FaceID = 2,
+  // Android iris authentication
+  Iris = 3,
+}
+
+export interface Status {
+  isAvailable: boolean;
+  biometryType: BiometryType;
+  error?: string;
+  errorCode?:
+    | "appCancel"
+    | "authenticationFailed"
+    | "invalidContext"
+    | "notInteractive"
+    | "passcodeNotSet"
+    | "systemCancel"
+    | "userCancel"
+    | "userFallback"
+    | "biometryLockout"
+    | "biometryNotAvailable"
+    | "biometryNotEnrolled";
+}
+
+export interface AuthOptions {
+  allowDeviceCredential?: boolean;
+  cancelTitle?: string;
+
+  // iOS options
+  fallbackTitle?: string;
+
+  // android options
+  title?: string;
+  subtitle?: string;
+  confirmationRequired?: boolean;
+  maxAttemps?: number;
+}
+
+export async function checkStatus(): Promise<Status> {
+  return invoke("plugin:biometric|status");
+}
+
+export async function authenticate(
+  reason: string,
+  options?: AuthOptions,
+): Promise<void> {
```

Comment:
Shouldn't this return whether it is successful or not? From my testings, it returns null on sucess and an object when cancelled

Reply:
it throws instead, added documentation for it